primitives/core/src/testing.rs

```diff
@@ -16,9 +16,15 @@
 
 //! Types that should only be used for testing!
 
+use crate::crypto::{KeyTypeId, CryptoTypePublicPair};
 #[cfg(feature = "std")]
-use crate::{ed25519, sr25519, crypto::{Public, Pair}};
-use crate::crypto::KeyTypeId;
+use crate::{
+	crypto::{Pair, Public},
+	ed25519, sr25519,
+	traits::BareCryptoStoreError
+};
+#[cfg(feature = "std")]
+use std::collections::HashSet;
 
 /// Key type for generic Ed25519 key.
 pub const ED25519: KeyTypeId = KeyTypeId(*b"ed25");
```

Comment:
Just noticed that we already have `KeyTypeId`, why do we introduce `CryptoTypeId` then?

Reply:
Oh, I guess `KeyTypeId` is actually "purpose" or "application", right? Can we please rename it to make it clear? Also This constant should most likely be renamed to something like `TEST_APP: CryptoAppId = CryptoAppId(*b"test");` to avoid confusion with `CryptoTypeId`.